src/commands/update.js

```diff
@@ -0,0 +1,122 @@
+var bot = require("../bot/bot.js").bot;
+var client = bot.get_client();
+var whois = require('./whois.js');
+
+const request = require('../util/request.js');
+
+function do_update(command, member, server_id) {
+    const serversTable = db.get_servers_table();
+    var guild = client.guilds.get(server_id);
+    Log.info(1, `Updating role for user: ${member.id}`);
+
+    new Promise((res, rej) =>
+            serversTable.ensure_exists()
+            .then(() => {
+                return serversTable.ensure_key_exists("server_id", server_id);
+            }).then(() => {
+                return serversTable.get_role_info(server_id);
+            })
+            .then((res) => {
+                res = res.rows[0].role_info;
+                var role_info = JSON.parse(res);
+                if (!role_info) {
+                    Log.error(1, "Returning %update because no role info");
+                    return;
+                }
+                var role_id;
+                whois.execute({guild: guild, channel:{send:function(){}}}, [`<@${member.id}>`], bot.db, {}).then((lichess) => {
+                    if (!lichess) {
+                        Log.error(1, "Update >> Could not fetch lichess username.");
+                        command.send_error("An unknown error has occurred.", "Have you registered your account?");
+                        return false;
+                    }
+                    request.user(lichess).then((user) => {
+                        var elo;
+                        if (user && user['perfs'] && user['perfs']['blitz'] && user['perfs']['blitz']['rating'])
+                            elo = parseInt(user['perfs']['blitz']['rating']);
+                        else
+                            elo = -1;
+                        for (var i = role_info.length-1; i >= 0; i--) {
+                            let role = role_info[i];
+                            if (role.from <= elo && role.to >= elo) {
+                                role_id = role.role_id;
+                            } else if (member.roles.has(role.id)) {
+                                member.removeRole(role.id).then(()=>{}).catch((e) => {
+                                    Log.error(1, "Role >> No permissions");
+                                    command.send_error("Hmm..", "I don't have the necessary permissions.");
+                                    return;
+                                });
+                                Log.error(1, `Role >> ${role.id} removed from ${member.id} (Updating role).`);
+                            }
+                        }
+                        if (!role_id) {
+                            Log.error(1, "Role >> No role applicable.");
+			    command.send_response("Update completed.", "Success");
+                            return;
+                        } else {
+                            member.addRole(role_id).then(() => {
+                                Log.error(1, "Role >> Role given (update)");
```

Comment:
error ?

Reply:
Honestly I copy and pasted a line without changing .error, oops.